users/views.py

```diff
@@ -124,6 +125,7 @@ def user_login(request):
     if request.user.is_authenticated():
         return redirect(next_page)
     if request.method == 'POST':
+        deactivate_non_constant_public_users()
```

Comment:
I don't think it's necessary to do this everytime someone want to login. It causes a lot QQ.
Can you do that after the user is authenticated?